### DISCUSSION TITLE
How to use wildcards in routeRules.

### DISCUSSION THREAD

**Discussion by TangsWang** [nuxt/nuxt#25004] [Questions]
### Environment

routeRules: {
    '/list/(.*).html?page=(.*)': { redirect: '/tags/$1/$2' },
  }

### Reproduction

routeRules: {
    '/list/(.*).html?page=(.*)': { redirect: '/tags/$1/$2' },
  }

### Describe the bug

routeRules: {
    '/list/(.*).html?page=(.*)': { redirect: '/tags/$1/$2' },
  }

### Additional context

routeRules: {
    '/list/(.*).html?page=(.*)': { redirect: '/tags/$1/$2' },
  }

**Comment by TheAlexLichter** [comment]
Right now this is not possible via routeRules, only with a custom nitro middleware + sendRedirect.

A first step would be when https://github.com/unjs/nitro/issues/1748 was resolved

**Comment by xjccc** [comment]
could support matcher?
```
routeRules: {
  '/xxx/c6_s93_([A-Z]+)': {
    redirect: ([, a]) => `/xxx/c11_s15_${a}`
  }
}
```

**Reply by TheAlexLichter** [reply]
@xjccc no, the content should be serializable, hence no functions are allowed in the config. But you could use e.g. $1, $2, ...

That'd be a discussion for https://github.com/unjs/nitro/issues/1748 though and not here.

**Comment by Yves852** [comment]
I'm facing the same issue and according to this april 2025 comment it may belong to Nuxt (3.16). https://github.com/nitrojs/nitro/issues/3303#issuecomment-2794294321

`'/brands/**': { redirect: { to: '/info/brands/**', statusCode: 301 } }`

The issue https://github.com/nuxt/nuxt/issues/23190 is closed and locked. Can I open a new issue?
